src/applications/app.helpers.js

```diff
@@ -36,15 +36,53 @@ export function isntLoaded(app) {
 
 export function shouldBeActive(app) {
   try {
-    return app.activeWhen(window.location);
+    if (Array.isArray(app.activeWhen))
```

Comment:
Oh! I after reading how you sanitize the app lifecycles, I agree to normalize how this should be done! Will move to the `sanitizeArguments` where I'm validating and transforming to correct internal api!
Another thing that I totally missed was the fact that some functions might be called a lot, and definitely perfomance tweaks should be considered! (will note on things to keep in mind when building a lib 👀)
I created a function which relies only on the parameters (and window) to reach it's result, so I could just move it outside this function... I actually moved inside because I thought that was a good way to say "look only `shouldBeActive` depends on `conditionForAppActivationMet`"